js/reference.js

```diff
@@ -376,65 +378,101 @@ var ERMrest = (function(module) {
 
                 var uri = this._uri;
 
-                // add sorting
-                // get a list of columns in key
-                // if table has no key, use all columns as key
-                var keycols;
-                if (this._table.keys.length() === 0) {
-                    keycols = this._table.columns.all();
-                } else {
-                    var keys = this._table.keys.all().sort( function(a, b) {
-                        return a.colset.length() - b.colset.length();
-                    });
-                    keycols = keys[0].colset.columns;
+                // if no sorting provided, use schema defined sort if that's present
+                // If neither one present, use shortestkey
+                if (!this._sort || (this._sort.length === 0)) {
+                    if (this._table.annotations.contains(module._annotations.TABLE_DISPLAY) &&
+                        this._table.annotations.get(module._annotations.TABLE_DISPLAY).contains("row_order")) {
```

Comment:
@hongsudt Here's something we haven't explicitly addressed. Notice `"row_order"` which is a key _within_ an annotation, in this case the table display annotation. Do we want those keys to also be from an enumeration? Might be `module._annotations.TABLE_DISPLAY_KEY_ROW_ORDER`? I don't feel strongly about this but I don't mind if we add it to the enumerations either.